web/client/epics/catalog.js

```diff
@@ -346,6 +349,31 @@ export default (API) => ({
                         );
                     }
                 }
+                if (layer.type === 'arcgis') {
+                    return Rx.Observable.defer(() => getLayerMetadata(layer.url, layer.name))
+                        .switchMap((properties) => {
+                            let esriProjectionId = properties?.extent?.spatialReference?.latestWkid || properties?.extent?.spatialReference?.wkid;
+                            let minP = CoordinatesUtils.reproject([properties.extent.xmin, properties.extent.ymin], `EPSG:${esriProjectionId}`, 'EPSG:4326');
+                            let maxP = CoordinatesUtils.reproject([properties.extent.xmax, properties.extent.ymax], `EPSG:${esriProjectionId}`, 'EPSG:4326');
+                            const newLayer = {
+                                ...layer,
+                                bbox: {
+                                    bounds: {
+                                        minx: minP?.x,
+                                        miny: minP?.y,
+                                        maxx: maxP?.x,
+                                        maxy: maxP?.y
+                                    },
+                                    crs: 'EPSG:4326'
+                                },
+                                // include metadata for future improvements
+                                properties
+                            };
+                            return Rx.Observable.from([addNewLayer({...newLayer, id})]);
```

Comment:
we could instead trigger the sequence of action we need like
```suggestion
                            return Rx.Observable.from([addNewLayer({...newLayer, id}), zoomToExtent(newLayer.bbox.bounds, newLayer.bbox.crs)]);
```

Reply:
Added here https://github.com/geosolutions-it/MapStore2/pull/10293/commits/3da19898fbd8ddfbd249392a27b25306277b3350